src/app/digital/public/api/DigitalPort.ts

```diff
@@ -8,4 +8,11 @@ export interface DigitalPort extends Port {
 
     readonly isInputPort: boolean;
     readonly isOutputPort: boolean;
+
+    /**
+     * Returns true if a port is available, false otherwise.
```

Comment:
Let's be more specific about what "available" means, i.e. explain the difference b/t input and output ports and what makes them "available"